src/objects/zcl_abapgit_object_msag.clas.abap

```diff
@@ -179,17 +180,82 @@ CLASS ZCL_ABAPGIT_OBJECT_MSAG IMPLEMENTATION.
 
 
   METHOD zif_abapgit_object~delete.
+    DATA: lv_t100a          TYPE t100a,
+          lv_error          TYPE abap_bool,
+          lv_frozen            TYPE flag,
+          lv_message_id        TYPE arbgb,
+          lv_trkey1            TYPE trkey,
+          lv_access_granted TYPE abap_bool.
+
+* parameter SUPPRESS_DIALOG doesnt exist in all versions of FM RS_DELETE_MESSAGE_ID
+* replaced with a copy
+    lv_message_id = ms_item-obj_name.
+    IF ms_item-obj_name EQ space.
+      lv_error = abap_true."blank message id
+    ELSE.
+      SELECT SINGLE * FROM t100a INTO lv_t100a WHERE arbgb = ms_item-obj_name.
+      IF sy-subrc NE 0.
+        lv_error = abap_true."not found
+      ENDIF.
+    ENDIF.
 
-* parameter SUPPRESS_DIALOG doesnt exist in all versions
-    CALL FUNCTION 'RS_DELETE_MESSAGE_ID'
-      EXPORTING
-        nachrichtenklasse = ms_item-obj_name
-      EXCEPTIONS
-        not_executed      = 1
-        not_found         = 2
-        no_permission     = 3
-        OTHERS            = 4.
-    IF sy-subrc <> 0.
+    IF lv_error = abap_false.
+      CLEAR lv_frozen.
+      CALL FUNCTION 'RS_ACCESS_PERMISSION'
+        EXPORTING
+          authority_check = 'X'
+          global_lock     = 'X'
+          mode            = 'MODIFY'
+          object          = lv_message_id
+          object_class    = 'T100'
+        IMPORTING
+          frozen          = lv_frozen
+        EXCEPTIONS
+          OTHERS          = 1.
+
+      IF sy-subrc NE 0 OR lv_frozen NE space.
+        lv_error = abap_true."can't access
+      ELSE.
+        lv_access_granted = abap_true.
+      ENDIF.
+
+    ENDIF.
+
+    IF lv_error = abap_false.
+
+      CALL FUNCTION 'RS_CORR_INSERT'
+        EXPORTING
+          global_lock        = 'X'
+          object             = lv_message_id
+          object_class       = 'MSAG'
+          mode               = 'D'
+        IMPORTING
+          transport_key      = lv_trkey1
+        EXCEPTIONS
+          cancelled          = 01
+          permission_failure = 02.
+      IF sy-subrc NE 0.
+        lv_error = abap_true."can't access
```

Comment:
raise exception?

Reply:
Done
was raising it at the end of the FM so I could use only one call to RS_ACCESS_PERMISSION with FREE (does a dequeue ).
Refactored to raise exceptions directly and call it in 2 places

I did consider to put it in a try...cleanup but never used it...